plugins/workspace-minimap/src/focus_mode.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A positionable version of the minimap.
+ * @author cesarades@google.com (Cesar Ades)
+ */
+
+import * as Blockly from 'blockly/core';
+
+const BlockEvents = new Set([
+  Blockly.Events.VIEWPORT_CHANGE,
+  Blockly.Events.BLOCK_CHANGE,
+  Blockly.Events.BLOCK_CREATE,
+  Blockly.Events.BLOCK_DELETE,
+  Blockly.Events.BLOCK_DRAG,
+  Blockly.Events.BLOCK_MOVE]);
+
+/**
+ * A class that highlights the user's viewport on the minimap.
+ */
+export class FocusMode {
+    private primaryWorkspace: Blockly.WorkspaceSvg;
+    private minimapWorkspace: Blockly.WorkspaceSvg;
+    private onChangeWrapper: (e: Blockly.Events.Abstract) => void;
+    private svgGroup: SVGElement;
+    private rect: SVGElement;
+    private background: SVGElement;
+    private id: string;
+
+
+    /**
+     * Constructor for focus mode.
+     * @param primary The primary workspaceSvg.
+     * @param minimap The minimap workspaceSvg.
+     */
+    constructor(primary: Blockly.WorkspaceSvg, minimap: Blockly.WorkspaceSvg) {
+      this.primaryWorkspace = primary;
+      this.minimapWorkspace = minimap;
+      this.id = this.minimapWorkspace.id;
+    }
+
+
+    /**
+     * Initializes focus mode.
+     */
+    init() {
+      // Make the svg group element.
+      this.svgGroup = Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.G, {'class': 'focusMode'}, null);
+
+      // Make the mask under the svg group.
+      const mask = Blockly.utils.dom.createSvgElement(
+          new Blockly.utils.Svg('mask'),
+          {'id': 'focusModeMask' + this.id},
+          this.svgGroup);
+
+      // Make the backround under the svg group.
+      this.background = Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'width': '100%',
+            'height': '100%',
+            'mask': 'url(#focusModeMask' + this.id + ')',
+          }, this.svgGroup);
+
+      // Make the white layer under the svg mask.
+      Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'width': '100%',
+            'height': '100%',
+            'fill': 'white',
+          }, mask);
+
+      // Make the black layer under the mask.
+      this.rect = Blockly.utils.dom.createSvgElement(
+          Blockly.utils.Svg.RECT, {
+            'x': 0,
+            'y': 0,
+            'rx': Blockly.Bubble.BORDER_WIDTH,
+            'ry': Blockly.Bubble.BORDER_WIDTH,
```

Comment:
Is this what the content highlight plugin is using? Either way, I would probably just define a constant in this class :P cuz this is not what that value was intended to be used for haha.

Reply:
Yea it is.
I'll make it a constant.